django/core/management/__init__.py

```diff
@@ -124,7 +124,8 @@ def call_command(command_name, *args, **options):
     args = defaults.pop('args', ())
     if 'skip_checks' not in options:
         defaults['skip_checks'] = True
-
+    if defaults.get('noinput', False) and not defaults.get('interactive', True):
+        raise CommandError('interactive=False should be passed rather noinput=True')
```

Comment:
Three things.

First I think a more appropriate solution would be to raise a `TypeError`.

Second it shouldn't special case the `noinput` and `interactive` cases but compare `kwargs` to the parser arguments's `dest`s.

Third It would be very neat if the `TypeError`s message would mention the allowed `kwargs` just like `filter()` does when you try to reference invalid fields.